### PR TITLE
Kev branch

### PR DESCRIPTION
**Pull request by kjzkcz**
# Pull Request Template

## Description

[comment]: <Finished the for loops, working on the two newline/elif/else problems> (Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change?)

## Type of change

[comment]: <New Feature> (Please delete options that are not relevant.)

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

[comment]: <Tested using the single line for loop codes in the test python file lines 41, 45> (Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

## Checklist:

- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ANTLR-version/App.g4 [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // Define grammar
 grammar App;
-s : comment (s)? | if_loop (s)? | definition (s)? | NEWLINE (s)? | while_loop (s)? | function (s)? | EOF  ;
+s : comment (s)? | if_loop (s)? | definition (s)? | NEWLINE (s)? | while_loop (s)? | function (s)? | for_loop (s)? | EOF  ;
 
-while_loop: WHILE ('(')? conditional_statement (')')? ':' s END;
+while_loop: WHILE ('(')? conditional_statement ('(')? ':' s END;
```
Comment (YoungBrendanM) [inline ANTLR-version/App.g4]: why did you make this change?

File: ANTLR-version/App.g4 [hunk 2]
```diff
@@ -15,13 +15,18 @@ conditional_statement: equation CONDITION equation (CONJOIN_CONDITION conditiona
 value : ID | NUMBER | STRING | BOOLEAN;
 equation : value | value ARITH_OPERATORS equation;
 
+for_loop: FOR value IN ('(')? function ('(')? ':' s END;
```
Comment (YoungBrendanM) [inline ANTLR-version/App.g4]: should there be a close paranthesis

File: ANTLR-version/App.g4 (hunk 1 above)
Comment (crsellinger) [inline ANTLR-version/App.g4]: change parentheses back around

File: ANTLR-version/antlr4.bat [hunk 3]
```diff
@@ -0,0 +1 @@
+java org.antlr.v4.Tool %*
```
Comment (YoungBrendanM) [inline ANTLR-version/antlr4.bat]: You should remove the bat files.

File: ANTLR-version/App.g4 (hunk 1 above)
Comment (crsellinger) [reply ANTLR-version/App.g4]: i saw the same thing. We're up to date with the latest commit

File: ANTLR-version/antlr4.bat (hunk 3 above)
Comment (crsellinger) [reply ANTLR-version/antlr4.bat]: Done

**Review by crsellinger — APPROVED** [review]

**Pull request merged**